src/components/instance-form-builder/InstanceFormBuilder.jsx

```diff
@@ -35,6 +36,20 @@ class InstanceFormBuilder extends React.Component {
         };
     }
 
+    componentDidMount = async () => {
+        const {
+            match: { params },
+            appConfig: { encryptionKey },
+            d2,
+        } = this.props;
+        const { instance, isEdit } = this.state;
+        if (isEdit && !instance.data.id) {
```

Comment:
I see we have a `Instance.get` and `Instance.getOrCreate`, one with d2 but no encryption key, the other with encryption key but not d2 (so it does queries the DB). Ideally, we should have a single `async Instance.getOrCreate(d2, id, encryptionKey)`. Then this code would be reduced to a single line.

Reply:
I agree, it's inconsistent this way and a bit tangled up. I'm going to take a look at it

Reply:
@SferaDev has asked to refactor this so he'll do it in another task